multimodal/src/autogluon/multimodal/utils/export.py

```diff
@@ -160,23 +152,85 @@ def export_onnx(
             else:
                 batch[key] = inp.to(device)
         self._model.to(device)
-        input_vec = InputBatch(**batch)
+        keys_to_drop = list(set(batch.keys()).difference(set(self._model.input_keys)))
+        for k in keys_to_drop:
+            batch.pop(k)
+        # Truncate input data types for TensorRT (only support: bool, int32, half, float)
+        if truncate_long_and_double:
+            for k in batch:
+                if batch[k].dtype == torch.int64:
+                    batch[k] = batch[k].to(torch.int32)
+        input_vec = [batch[k] for k in self._model.input_keys]
+
+        self._model.eval()
+
+        if not path:
+            onnx_path = os.path.join(self.path, "model.onnx")
+        else:
+            onnx_path = os.path.join(path, "model.onnx")
+        dirname = os.path.dirname(os.path.abspath(onnx_path))
+        if not os.path.exists(dirname):
+            os.makedirs(dirname)
 
-        with precision_context, torch.no_grad():
-            traced_model = torch.jit.trace(self._model, input_vec)
-        torch.jit.save(traced_model, "traced_model.pt")
-        traced_model = torch.jit.load("traced_model.pt")
+        input_keys = self._model.input_keys
```

Comment:
Consider putting this line at the beginning of this function. I see some places above uses `self._model.input_keys`. Using one variable can make code more organized.

Reply:
placed to the top location, and the local input_keys variable is reused.